rocketpy/Function.py

```diff
@@ -473,6 +473,160 @@ def setDiscrete(
             self.__interpolation__ = "shepard"
         return self
 
+    def setDiscreteBasedOnModel(self, modelFunction, oneByOne=True):
+        """This method transforms a Function instance defined from callables into a
+        Function instance defined by a list of discrete points.
```

Comment:
The method also works with discrete based Functions right?

So it also transforms the mesh of a Function instance defined by discrete points into the mesh of the modelFunction

Reply:
Perfect observation, I have adjusted in the last commit!